Git worktrees don't work

**Describe the bug**

Kiwix-android's custom Gradle scripts crash when building inside a Git worktree.

**Expected behavior**

It should build whether in a worktree or not. It should copy to the main `.git` directory as hooks are inherited by worktrees.

**Steps to reproduce the behavior:**

```bash
gradle clean
gradle --stop
git worktree add ../java19 fix/java19
cd ../java19
gradle build
```

**Environment**
- Version of Kiwix Android : Local Git work after e19bb9cf978ea628935df5f777fc02748b62cc32
- OS version : Ubuntu 22.10

**Logs**

```
$ gradle build

> Configure project :app
[directory '/home/home/AndroidStudioProjects/java19/app/build', directory '/home/home/AndroidStudioProjects/java19/core/build', directory '/home/home/AndroidStudioProjects/java19/custom/build']WARNING:BuildType(playStore): buildConfigField 'IS_PLAYSTORE' value is being replaced.

> Task :core:copyGitHooks FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':core:copyGitHooks' (type 'Copy').
  - Type 'org.gradle.api.tasks.Copy' property 'destinationDir' is not writable because '/home/home/AndroidStudioProjects/java19/.git/hooks' ancestor '/home/home/AndroidStudioProjects/java19/.git' is not a directory.
    
    Reason: Expected '/home/home/AndroidStudioProjects/java19/.git' to be a directory but it's a file.
    
    Possible solution: Make sure that the 'destinationDir' is configured to a directory.
    
    Please refer to https://docs.gradle.org/7.6/userguide/validation_problems.html#cannot_write_output for more details about this problem.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 657ms
7 actionable tasks: 1 executed, 6 up-to-date
```

I have a solution based on my answer at https://stackoverflow.com/a/75780328/10477326 , and will open a PR shortly.

This affects more than just worktrees:

```
Executing tasks: [uninstallAll] in project /home/home/AndroidStudioProjects/kiwix-android/app

> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:inspectClassesForKotlinIC UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:assemble UP-TO-DATE
> Task :buildSrc:compileTestKotlin NO-SOURCE
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:pluginUnderTestMetadata UP-TO-DATE
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:validatePlugins UP-TO-DATE
> Task :buildSrc:check UP-TO-DATE
> Task :buildSrc:build UP-TO-DATE

> Configure project :app
[directory '/home/home/AndroidStudioProjects/kiwix-android/app/build', directory '/home/home/AndroidStudioProjects/kiwix-android/core/build', directory '/home/home/AndroidStudioProjects/kiwix-android/custom/build']AGPBI: {"kind":"warning","text":"BuildType(playStore): buildConfigField 'IS_PLAYSTORE' value is being replaced.","sources":[{}]}

> Task :app:copyGitHooks

> Task :app:installGitHooks FAILED
chmod: cannot access '../.git/hooks': No such file or directory

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:installGitHooks'.
> Process 'command 'chmod'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s
8 actionable tasks: 2 executed, 6 up-to-date
```

My PR previously only fixed the case for worktrees, but some existing things are still broken like previously. I force-pushed it to use absolute paths so that certain Gradle tasks in Android Studio will start working now.

--- a/team-props/git-hooks.gradle
+++ b/team-props/git-hooks.gradle
@@ -3,13 +3,15 @@ static def isLinuxOrMacOs() {
   return osName.contains('linux') || osName.contains('mac os') || osName.contains('macos')
 }
 
+def gitHooksDir = 'git rev-parse --path-format=absolute --git-path hooks'.execute().text.trim()
+
 task copyGitHooks(type: Copy) {
   description 'Copies the git hooks from team-props/git-hooks to the .git folder.'
   from("${rootDir}/team-props/git-hooks/") {
     include '**/*.sh'
     rename '(.*).sh', '$1'
   }
-  into "${rootDir}/.git/hooks"
+  into gitHooksDir
 }
 
 task installGitHooks(type: Exec) {
@@ -17,7 +19,7 @@ task installGitHooks(type: Exec) {
   group 'git hooks'
   workingDir rootDir
   commandLine 'chmod'
-  args '-R', '+x', '.git/hooks/'
+  args '-R', '+x', gitHooksDir
   onlyIf { isLinuxOrMacOs() }
   dependsOn copyGitHooks
   doLast {

